src/pcm-sensor-server.cpp

```diff
@@ -3726,6 +3726,7 @@ void printHelpText( std::string const & programName ) {
 #if defined (USE_SSL)
     std::cout << "    -s                   : Use https protocol (default port " << DEFAULT_HTTPS_PORT << ")\n";
 #endif
+    std::cout << "    -l bind ip6          : Bind to ip6 address. (default ip6 is ::)\n";
```

Comment:
I still prefer using -i and or use a long option --ipv6 or so. What about ipv4? Why only ipv6?

Reply:
Server class uses dual IP stack properties on Linux, no need to be specific.

Anyway you use a string to pass it around, what refactoring would that require?

But before we continue this, why exactly do you need to pass an IPv6 address in here? By default it should take all addresses, ipv4 and ipv6, is this not working for you?

Reply:
As you said, currently the server listens on `::`. This is working.
I'd like to specify the address the server is listening on. To the best of my knowledge I would need to specify AF_INET to listen on a specific v4 address or AF_INET6 to listen on a specific v6 address. Currently the `Server` class is using AF_INET6 internally.

Reply:
IPv4 addresses can be represented as ipv6 addresses, see RFC 4291 for details but it looks like ::ffff:d.d.d.d, d.d.d.d is the ipv4 address just like normal using decimal format, not hexadecimal like ipv6. This would be one way to make it listen on an ipv4 address while still using AF_INET6 but i am not sure this is needed. Going the route of ipv4 mapped ipv6 addresses means the URL IPv6 validator needs to be fixed, it does not account for this format. That is an oversight on my part when writing that IPv6 validation in URL.

Reply:
Caveat: I have not tested my IPv6 code a system that does not support IPv6 (are there still some around?) so i do not know what :: using AF_INET6 does on uch a system that can only handle IPv4...